### PR TITLE
HPCC-28155 ESDL WSDL cli should recognize ESDL option use_method_name

### PR DESCRIPTION
**Pull request by afishbeck**
WSDLs are slightly different for services with use_method_name set.

Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: esp/xslt/esxdl2xsd.xslt [hunk 1]
```diff
@@ -359,14 +359,15 @@
...
+        <xsl:variable name="useMethodName" select="@use_method_name='1'"/>
         <wsdl:message name="EspSoapFault">
             <wsdl:part name="parameters" element="tns:Exceptions"/>
         </wsdl:message>
         <xsl:for-each select="/esxdl/EsdlMethod">
             <wsdl:message>
                 <xsl:attribute name="name"><xsl:value-of select="@name"/>SoapIn</xsl:attribute>
                 <wsdl:part name="parameters">
-                    <xsl:attribute name="element">tns:<xsl:value-of select="@request_type"/></xsl:attribute>
+                    <xsl:attribute name="element">tns:<xsl:value-of select="@name"/><xsl:if test="not($useMethodName)">Request</xsl:if></xsl:attribute>
```
Comment (rpastrana) [inline esp/xslt/esxdl2xsd.xslt:370]: This part looks fine as long as our requests are guaranteed to be labeled `<methodname>Request` - using `@name` or `@request_type` based on useMethodName seems slightly less problematic.

But more importantly, it seems the xsd:element definition referenced by this attribute needs to recognize useMethodName as well.

**Review by rpastrana — CHANGES_REQUESTED** [review]

File: esp/xslt/esxdl2xsd.xslt (hunk 1 above)
Comment (rpastrana) [reply esp/xslt/esxdl2xsd.xslt:370]: @afishbeck it seems the EsdlRequest template (L:338) has no direct relationship to the method...
How do you feel about inferring the method name from the requestname?

`<xsl:attribute name="name"><xsl:value-of select="@name"/></xsl:attribute>`
to
`<xsl:attribute name="name><xsl:if test="$useMethodName"><xsl:value-of select="substring-before(@name, 'Request')"/></xsl:if><xsl:else><xsl:value-of select="@name"/></xsl:else></xsl:attribute>`

Comment (rpastrana) [reply esp/xslt/esxdl2xsd.xslt:370]: @afishbeck bump

Comment (afishbeck) [reply esp/xslt/esxdl2xsd.xslt:365]: @rpastrana updated.  I basically did what you suggested, just had to update the syntax a bit.

Comment (afishbeck) [reply esp/xslt/esxdl2xsd.xslt:365]: @rpastrana Are you able to test?

**Comment by rpastrana** [comment]
@Michael-Gardner this is the PR we discussed, please test the xslt against your process. Thanks.

**Review by rpastrana — APPROVED** [review]
@afishbeck looks fine, awaiting @Michael-Gardner 's confirmation

**Comment by rpastrana** [comment]
@Michael-Gardner please post your findings here.

**Comment by Michael-Gardner** [comment]
I'm not getting the name mismatch in the output of my `esdl wsdl` commands now. Seems to match what was previously expected.

**Review by Michael-Gardner — APPROVED** [review]

**Pull request merged**